linkerd/app/outbound/src/ingress.rs

```diff
@@ -130,13 +131,10 @@ where
         .into_inner();
 
     svc::stack(http::NewServeHttp::new(h2_settings, http, tcp, drain))
-        .check_new_service::<tcp::Accept, io::PrefixedIo<transport::metrics::SensorIo<I>>>()
-        .push_on_response(svc::layers().push_spawn_buffer(buffer_capacity).push(
-            transport::Prefix::layer(
-                http::Version::DETECT_BUFFER_CAPACITY,
-                detect_protocol_timeout,
-            ),
-        ))
+        .check_new_service::<(Option<http::Version>, tcp::Accept), io::PrefixedIo<transport::metrics::SensorIo<I>>>()
+        .check_new_clone::<(Option<http::Version>, tcp::Accept)>()
```

Comment:
nit, take it or leave it: i wonder if the `(Option<http::Version>, tcp::Accept)` tuple ought to be a named struct, or at least a type alias?

also, maybe we could import `transport::metrics::SensorIo` so this line is a little less wordy? up to you, though.

Reply:
i'm okay with using tuple targets temporarily -- i think it's better than having a bunch of strict dependencies and traits. For instance, ServeHttp takes `(Option<Version>, T)` and uses `(Version, T)` on its inner service... doing this with named types/traits would be much more verbose for no real benefit, imo

Reply:
ah, yup, that makes sense --- carry on then!